inc/Symbol.h

```diff
@@ -0,0 +1,46 @@
+#ifndef SYMBOL_H
+#define SYMBOL_H
+
+#include <iostream>
+
+namespace Calculator {
+
+/**
+ * @brief Represents a symbol in a mathematical expression. Symbols can be any single character from a-z.
+ * 
+ */
+template<typename T>
+class Symbol {
+public:
+    const char symbol;
```

Comment:
no public members are allowed

private members should have a trailing underscore, e.g. `symbol_`. I don't make the rules 🤷